### ISSUE TITLE
Port to v0.17

### ISSUE THREAD

**Issue by amitaibu** [Gizra/elm-hedley#145]

**Comment by amitaibu** [comment]
Waiting for https://github.com/rgrempel/elm-route-hash/issues/13#issuecomment-224367690

**Comment by michallepicki** [comment]
I would love to see this ported to v0.17. Looks like a great learning resource!

**Comment by amitaibu** [comment]
Indeed, just missing some time. Hopefully will get it going soon with other Gizra devs.

**Comment by DavidHernandez** [comment]
I'm working on this. elm-route-url already supports 0.17 and there are no more blocking dependencies.

@amitaibu, if you give me access to the repo I will push a new branch with my current changes so anyone else can contribute.

**Comment by DavidHernandez** [comment]
There are some packages that were not being handled using the elm package manager, instead they are directly on a vendor folder. One of them at least, is not going to be upgraded to 0.17: https://github.com/rgrempel/elm-web-api/issues/17

I'm investigating if this vendor is really necessary or if we can get rid of it.

**Comment by michallepicki** [comment]
Only Location seems to be used, so that should be easy. I found two modules that are using it:
[Config/Update.elm](https://github.com/Gizra/elm-hedley/blob/2a28e1abd907cca93ab3f843f19ed31cc5e85e97/src/elm/Config/Update.elm) and [Pages/GithubAuth/Update.elm](https://github.com/Gizra/elm-hedley/blob/2a28e1abd907cca93ab3f843f19ed31cc5e85e97/src/elm/Pages/GithubAuth/Update.elm)

**Comment by lisardo** [comment]
@DavidHernandez did you have any progress? I would love to see a port to 0.17!

**Comment by DavidHernandez** [comment]
Not really, and I actually lost some of the progress as my laptop was stolen :(

I will try to allocate some time during the next few weeks to work on this.

**Comment by DavidHernandez** [comment]
I'm starting to work on this again. Following the original proposal from @amitaibu, I've removed everything I will start from the begining. All the work will be done on the same PR, on this branch from my fork: https://github.com/DavidHernandez/elm-hedley/tree/0.17-upgrade

I will go replicating each of the existing pages. I already started with the login. If anyone wants to help, just fork my repo and do PRs.

**Comment by amitaibu** [comment]
👍

**Comment by DavidHernandez** [comment]
As 0.18 has been released, so we will go directly for the 0.18 upgrade.

**Comment by nader77** [comment]
@DavidHernandez What's the status on this ? Do you need any help ?

**Comment by DavidHernandez** [comment]
@nader77 right now there is only one vendor library left with no support for elm 0.18 and it is https://github.com/rgrempel/elm-route-url

I will open an issue on that repository about upgrading to 0.18.

**Comment by amitaibu** [comment]
There's already a PR for that -- https://github.com/rgrempel/elm-route-url/pull/19

**Comment by DavidHernandez** [comment]
By the way, to work with different versions of Elm, there is this npm module that does a nice job: https://www.npmjs.com/package/elm-version-manager
